datadog_checks_base/datadog_checks/base/utils/db/utils.py

```diff
@@ -158,6 +159,25 @@ def default_json_event_encoding(o):
     raise TypeError
 
 
+def obfuscate_sql_with_metadata(query, options=None):
+    try:
+        statement = datadog_agent.obfuscate_sql(query, options)
+        statement_with_metadata = json.loads(statement)
+    except json.JSONDecodeError:
+        # Assume we're running against an older agent and return the obfuscated query without metadata.
+        return {'query': statement, 'metadata': {}}
+    except Exception as e:
+        raise e
+
+    metadata = statement_with_metadata.get('metadata', {})
+    tables = metadata.pop('tables_csv', None)
+    # The obfuscator will at least return an empty string. We need to check for this because we want to
+    # omit None values later down the pipeline and splitting would result in [''].
+    tables = tables.split(',') if tables else None
```

Comment:
Should also probably strip whitespace and exclude empty strings from the result post-split.